src/pages/Schedule/Filter.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faArrowDown } from '@fortawesome/free-solid-svg-icons';
+import * as S from './StyledSchedule';
+import styled from 'styled-components';
+
+const Filter = ({ dropDownClicked, isBtnOpen, handleFilter }) => {
+  return (
+    <>
+      {FILTER_NAME && (
+        <S.Box>
+          <span onClick={() => dropDownClicked('first')}>Name</span>
+          <S.CheckIcon isBtnOpen={isBtnOpen}>
+            <StyledFontAwesomeIcon
+              onClick={() => dropDownClicked('first')}
+              icon={faArrowDown}
+            />
+          </S.CheckIcon>
+        </S.Box>
+      )}
+
+      {FILTER_LEVEL && (
+        <S.Box>
+          <span onClick={() => dropDownClicked('second')}>Level</span>
+          <S.CheckIcon isBtnOpen={isBtnOpen}>
+            <StyledFontAwesomeIcon
+              onClick={() => dropDownClicked('second')}
+              icon={faArrowDown}
+            />
+          </S.CheckIcon>
+        </S.Box>
+      )}
+
+      {isBtnOpen?.first && (
+        <S.FilterBox isBtnOpen={isBtnOpen}>
+          {FILTER_NAME.map((name, idx) => (
+            <S.FilterItem
+              key={idx}
+              onClick={e => handleFilter(e)}
+              value={name}
+              id="1"
+            >
+              {name}
+            </S.FilterItem>
+          ))}
+        </S.FilterBox>
+      )}
+
+      {isBtnOpen?.second && (
+        <S.FilterBox isBtnOpen={isBtnOpen}>
+          {FILTER_LEVEL.map((name, idx) => (
+            <S.FilterItem
+              key={idx}
+              onClick={e => handleFilter(e)}
+              value={name}
+              id="2"
+            >
+              {name}
+            </S.FilterItem>
+          ))}
+        </S.FilterBox>
+      )}
+    </>
+  );
+};
```

Comment:
```suggestion
const Dropdown = ({ name, type }) => {
  const [isDropdownClick, setIsDropdownClick] = useState(false);

  const dropdownClicked = () => {
    setIsDropdownClick(prev => !prev);
  };

  return (
    <>
      <S.Box onClick={dropdownClicked}>
        <span>{type}</span>
        <S.CheckIcon isDropdownClick={isDropdownClick}>
          <StyledFontAwesomeIcon icon={faArrowDown} />
        </S.CheckIcon>
      </S.Box>
      {isDropdownClick && (
        <S.FilterBox>
          {name.map((name, idx) => (
            <S.FilterItem key={idx} onClick={dropdownClicked}>
              {name}
            </S.FilterItem>
          ))}
        </S.FilterBox>
      )}
    </>
  );
};
```

하나의 컴포넌트 안에서 버튼을 눌렀을 때 나오는 드롭다운 내용을 위와 같은 방법으로
하나의 state로 관리할 수 있습니다.
이렇게 하나의 컴포넌트를 만들어 두고, 부모에서 props로 데이터를 넘겨주는 방식으로 구현하면
조금 더 효율적으로 UI를 관리할 수 있게 됩니다.

잘 이해가 되지 않는다면 세션 중에 '컴포넌트 재사용' 세션을 복기 해보면
어떤 내용을 말하는지 아실 수 있겠습니다.

Reply:
명성님 차분하게 다시 공부해서 완벽히 이해하겠습니다!! 감사합니다!